src/SSHDebugPS/Docker/DockerContainerInstance.cs

```diff
@@ -13,20 +13,21 @@ public class DockerContainerInstance : ContainerInstance
         /// <summary>
         /// Create a DockerContainerInstance from the results of docker ps in JSON format
         /// </summary>
-        public static DockerContainerInstance Create(string json)
+        public static bool TryCreate(string json, out DockerContainerInstance instance, out string error)
         {
+            instance = null;
+            error = string.Empty;
             try
             {
                 JObject obj = JObject.Parse(json);
-                var instance = obj.ToObject<DockerContainerInstance>();
-                if (instance != null)
-                    return instance;
+                instance = obj.ToObject<DockerContainerInstance>();
             }
             catch (Exception e)
             {
-                Debug.Fail(e.ToString());
+                error = e.ToString();
+                Debug.Fail(error);
             }
```

Comment:
>} [](start = 12, length = 1)

Are you sure you don't want to do the logging in the spot where you have the exception so that you can also include some part of the failure info?